src/sage/combinat/combinat.py

```diff
@@ -2974,17 +2982,14 @@ def bell_polynomial(n: Integer, k: Integer):
     - Thierry Monteil (2015-09-29): the result must always be a polynomial.
     """
     from sage.combinat.partition import Partitions
-    R = PolynomialRing(ZZ, 'x', n - k + 1)
+    R = PolynomialRing(ZZ, 'x0' if n == k else 'x', n - k + 1)
```

Comment:
```suggestion
    if n == k:
        R = PolynomialRing(ZZ, 'x0')
    else:
        R = PolynomialRing(ZZ, 'x', n - k + 1)
```
This is easier to understand and the univariate case then knows it is univariate (with more options).